test/test_unary_ufuncs.py

```diff
@@ -295,7 +295,7 @@ def test_reference_numerics(self, device, dtype, op):
                                            exact_dtype=exact_dtype, rtol=1e-3, atol=1e-2)
                     continue
 
-            self.assertEqualHelper(actual, expected, msg, dtype=dtype, exact_dtype=exact_dtype)
+            self.assertEqualHelper(actual, expected, msg, dtype=dtype, equal_nan="relaxed", exact_dtype=exact_dtype)
```

Comment:
Does it make sense to do the following?
```python
if torch.isnan(t).any():
    self.assertEqualHelper(actual, expected, msg, dtype=dtype, equal_nan="relaxed", exact_dtype=exact_dtype)
else:
    self.assertEqualHelper(actual, expected, msg, dtype=dtype, exact_dtype=exact_dtype)
```
cc: @mruberry

Reply:
And before that
```python
        tensors = generate_numeric_tensors(device, dtype,
                                           domain=op.domain,
                                           include_large_values=op.handles_large_floats,
                                           include_extremal_values=False) + \
        generate_numeric_tensors(device, dtype,
                                           domain=op.domain,
                                           include_large_values=op.handles_large_floats,
                                           include_extremal_values=include_extremals)
```

Reply:
Why would the equal_nan setting depend on whether NaNs are present or not?

I'd be inclined not to take a change like this vs. focusing on splitting the reference numerics test up.